org.civicrm.reqangular/src/common/models/settings.model.js

```diff
@@ -0,0 +1,54 @@
+/* eslint-env amd */
+
+define([
+  'common/modules/models',
+  'common/models/model',
+  'common/models/instances/settings-instance',
+  'common/services/api/settings.api'
+], function (models) {
+  'use strict';
+
+  models.factory('Settings', [
+    '$log', 'Model', 'api.settings', 'SettingsInstance',
+    function ($log, Model, SettingsAPI, instance) {
+      $log.debug('Settings');
+
+      var promise;
+
+      return Model.extend({
+
+        /**
+         * Calls the get() method of the Settings API, and returns a
+         * SettingsInstance for each setting.
+         *
+         * @param  {Object} params
+         * @return {Promise}
+         */
+        get: function (params) {
+          return SettingsAPI.get(params)
+            .then(function (settings) {
+              return settings.map(function (setting) {
+                return instance.init(setting, true);
```

Comment:
we are creating a `SettingInstance` for each setting? why? i thought that an instance would represent the whole group of settings, what's the benefit of having an instance for each individual setting instead?

Reply:
Actually, `SettingsAPI.get(...)` returns the array of setting object results as:
```js
[
   {
      propertyOne: '',
      propertyTwo: '',
      ....
   }
]
```
So, it will represent the whole group of settings not the individual settings property.